static-analysis-plugin/plugin/src/main/groovy/com/novoda/staticanalysis/StaticAnalysisExtension.groovy

```diff
@@ -0,0 +1,27 @@
+package com.novoda.staticanalysis
+
+import org.gradle.api.Action
+
+class StaticAnalysisExtension {
```

Comment:
This models the extension of the plugin, and at the moment allows only to specify a `penalty` containing the  thresholds to be considered for warnings/errors before breaking the build. There are three (hope self-explanatory) built-in penalties (`none`, `failOnErrors`, `failOnWarnings`), but you could define your own thresholds if you like as:
```
staticAnalysis {
    penalty {
        maxWarnings 30
        maxErrors 5
    }
}
```